app/src/lib/stores/app-store.ts

```diff
@@ -1751,6 +1753,35 @@ export class AppStore extends TypedBaseStore<IAppState> {
     this._updateCurrentPullRequest(repository)
     this.updateMenuItemLabels(repository)
     this._initializeCompare(repository)
+    this.repositories = await this.getRefreshedRepositories(
+      this.repositories,
+      repository
+    )
+  }
+
+  public async refreshAllRepositories() {
+    this.repositories = await this.getRefreshedRepositories(this.repositories)
+    this.emitUpdate()
+  }
+
+  private async getRefreshedRepositories(
+    repositories: ReadonlyArray<Repository>,
+    targetRepo?: Repository
+  ): Promise<ReadonlyArray<Repository>> {
+    for (const repo of repositories) {
+      if (!targetRepo || targetRepo.id === repo.id) {
+        await this.withAuthenticatingUser(repo, async (repo, account) => {
+          const gitStore = this.getGitStore(repo)
+          repo.setAheadBehind(gitStore.aheadBehind)
```

Comment:
:art: the backing fields for `setAheadBehind` and `setChangedFiles` are currently public, so they can be set directly without the need for also having a function defined:

```ts
repo.aheadBehind = gitStore.aheadBehind
```

Alternatively, if you wanted to protect access to these you can use getters and setters. This is a TS example of the syntax:

```ts
class Employee {
    private _fullName: string;

    get fullName(): string {
        return this._fullName;
    }

    set fullName(newName: string) {
        if (passcode && passcode == "secret passcode") {
            this._fullName = newName;
        }
        else {
            console.log("Error: Unauthorized update of employee!");
        }
    }
}
```